hardware/raspberrypi/booteeprom.md

```diff
@@ -2,82 +2,118 @@
+### Reading the current EEPROM configuration
+To view the configuration file used by the bootloader at boot time run `rpi-eeprom-config` or `vcgencmd bootloader_config`.
+
+### Editing the current bootloader configuration
+The following command loads the current EEPROM configuration into a text editor. When the editor is closed, `rpi-eeprom-config` applies the updated configuration to latest available EEPROM release and uses `rpi-eeprom-update` to schedule an update when the system is rebooted:
 
 ```
-sudo rpi-eeprom-update
+sudo -E rpi-eeprom-config --edit
+sudo reboot
 ```
+If the updated configuration file is identical or empty then no changes are made.
 
-If an update is available, you can install it using :
+The editor is selected by the `EDITOR` environment variable.
 
+### Applying a saved configuration file
+The following command applies `boot.conf` to the latest available EEPROM image and uses `rpi-eeprom-update` to schedule an update when the system is rebooted.
 ```
-sudo rpi-eeprom-update -a
+sudo rpi-eeprom-config --apply boot.conf
 sudo reboot
 ```
 
-### Reading the current EEPROM configuration
+### Reading the configuration file from an EEPROM image
+To read the configuration file from an EEPROM image file:
+```
+rpi-eeprom-config pieeprom.bin
+```
+
+### Low level commands
 
-To view the configuration file used by the bootloader at boot time
+#### Updating the configuration in an EEPROM image
+The following command reads `pieeprom.bin` and replaces the configuration file with the contents of `boot.conf`. The result is written to `new.bin`
 ```
-vcgencmd bootloader_config
+rpi-eeprom-config --config boot.conf --out new.bin pieeprom.bin
 ```
 
-### Reading the EEPROM version
+For more information about advanced options please run `rpi-eeprom-config -h`.
+
+#### Updating the bootloader EEPROM
+The following will cause the bootloader EEPROM to be updated the next time the system is rebooted.
 ```
-vcgencmd bootloader_version
+# -d instructs rpi-eeprom-update to use the EEPROM configuration inside new.bin instead of
+# of replacing it with the current configuration from vcgencmd bootloader.
+sudo rpi-eeprom-update -d -f new.bin
 ```
+For more information about advanced options please run `rpi-eeprom-update -h`.
+
+### recovery.bin
+At power on, the BCM2711 ROM looks for a file called `recovery.bin` in the root directory of the boot partition on the sd-card. If a valid `recovery.bin` is found then the ROM executes this instead of the SPI EEPROM image. This mechanism ensures that the bootloader SPI EEPROM can always be reset to valid image with factory default settings.
+
+#### EEPROM update files
+| Filename | Purpose |
+|----------|---------|
+| recovery.bin | VideoCore EEPROM recovery executable |
+| pieeprom.upd | Bootloader EEPROM image |
+| pieeprom.bin | Bootloader EEPROM image (same as pieeprom.upd) |
```

Comment:
Might be worth clarifying that you'd normally only have `pieeprom.upd` OR `pieeprom.bin`, and not both at the same time?